cli/running/running.go

```diff
@@ -470,50 +460,65 @@ func FillCtx(cliOpts *config.CliOpts, cmdCtx *cmdcontext.CmdCtx,
-	if err != nil {
-		return fmt.Errorf("can't find an application init file: %s", err)
+	var appList []string
+	if len(args) == 0 {
+		appList, err = util.CollectAppList(instEnabledPath)
+		if err != nil {
+			return fmt.Errorf("can't collect an application list "+
+				"from instances enabled path: %s", err)
+		}
+	} else {
+		appList = append(appList, args[0])
 	}
 
-	// Cleanup instances list.
 	runningCtx.Instances = nil
-	for _, inst := range instParams {
-		var instance InstanceCtx
-		var runDir string
-		var logDir string
-		var dataDir string
-
-		instance.AppPath = inst.AppPath
-		instance.AppName = inst.AppName
-		instance.InstName = inst.InstName
-
-		if cliOpts.App != nil {
-			runDir = cliOpts.App.RunDir
-			logDir = cliOpts.App.LogDir
-			dataDir = cliOpts.App.DataDir
-			instance.LogMaxSize = cliOpts.App.LogMaxSize
-			instance.LogMaxAge = cliOpts.App.LogMaxAge
-			instance.LogMaxBackups = cliOpts.App.LogMaxBackups
-			instance.Restartable = cliOpts.App.Restartable
+	for _, appName := range appList {
+		if strings.HasSuffix(appName, ".lua") {
+			appName = appName[0 : len(appName)-4]
+		}
+		instParams, err := collectInstances(appName, cliOpts, instEnabledPath)
+		if err != nil {
+			return fmt.Errorf("%s: can't find an application init file: %s", appName, err)
 		}
 
-		instance.RunDir = makePath(runDir, basePath, &inst)
-		instance.ConsoleSocket = filepath.Join(instance.RunDir, instance.InstName+".control")
-		instance.PIDFile = filepath.Join(instance.RunDir, instance.InstName+".pid")
-
-		instance.LogDir = makePath(logDir, basePath, &inst)
-		instance.Log = filepath.Join(instance.LogDir, instance.InstName+".log")
-
-		instance.DataDir = makePath(dataDir, basePath, &inst)
+		// Cleanup instances list.
+		for _, inst := range instParams {
+			var instance InstanceCtx
+			var runDir string
+			var logDir string
+			var dataDir string
+
+			instance.AppPath = inst.AppPath
+			instance.AppName = inst.AppName
+			instance.InstName = inst.InstName
+
+			if cliOpts.App != nil {
+				runDir = cliOpts.App.RunDir
+				logDir = cliOpts.App.LogDir
+				dataDir = cliOpts.App.DataDir
+				instance.LogMaxSize = cliOpts.App.LogMaxSize
+				instance.LogMaxAge = cliOpts.App.LogMaxAge
+				instance.LogMaxBackups = cliOpts.App.LogMaxBackups
+				instance.Restartable = cliOpts.App.Restartable
+			}
 
-		if cmdCtx.CommandName != "run" {
-			err = createDataDir(instance.DataDir)
-			if err != nil {
-				return err
+			instance.RunDir = makePath(runDir, basePath, &inst)
+			instance.ConsoleSocket = filepath.Join(instance.RunDir, instance.InstName+".control")
+			instance.PIDFile = filepath.Join(instance.RunDir, instance.InstName+".pid")
+			instance.LogDir = makePath(logDir, basePath, &inst)
+			instance.Log = filepath.Join(instance.LogDir, instance.InstName+".log")
+			instance.DataDir = makePath(dataDir, basePath, &inst)
+			instance.SingleApp = inst.SingleApp
+
+			if cmdCtx.CommandName != "run" {
+				err = createDataDir(instance.DataDir)
```

Comment:
Suppose, this will create data dir, even if I execute `tt stop`.

Reply:
Why?

Reply:
Because "stop" != "run"

Reply:
Oops, I see now, thanks for noticing. Fixed.

Reply:
Pls make sure, "restart" command with all stopped apps still creates the data directory.